### PR TITLE
Separated tests in contenttypes_tests.

### PR DESCRIPTION
**Pull request by desecho**
Separated tests in contenttypes_tests (requested  by @timgraham) https://github.com/django/django/pull/7874#discussion_r98234831

### CODE REVIEW

File: tests/contenttypes_tests/test_checks.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    GenericForeignKey, GenericRelation,
+)
+from django.contrib.contenttypes.models import ContentType
+from django.core import checks
+from django.db import models
+from django.test import SimpleTestCase, override_settings
+from django.test.utils import isolate_apps
+
+
+@override_settings(SILENCED_SYSTEM_CHECKS=['fields.W342'])  # ForeignKey(unique=True)
```
Comment (charettes) [inline tests/contenttypes_tests/test_checks.py]: I think this is not required anymore now that these tests rely on `isolate_apps`?

File: tests/contenttypes_tests/test_checks.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+        expected = [
+            checks.Error(
+                'Field names must not end with an underscore.',
+                obj=Model.content_object_,
+                id='fields.E001',
+            )
+        ]
+        self.assertEqual(errors, expected)
+
+    @override_settings(INSTALLED_APPS=['django.contrib.auth', 'django.contrib.contenttypes', 'contenttypes_tests'])
```
Comment (charettes) [inline tests/contenttypes_tests/test_checks.py]: Should should be able to remove `'django.contrib.auth'` from there.

Comment (desecho) [inline tests/contenttypes_tests/test_checks.py]: If I remove it I get the following error:

```
======================================================================
ERROR: test_generic_foreign_key_checks_are_performed (contenttypes_tests.test_checks.GenericForeignKeyTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.4/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/usr/lib/python3.4/unittest/case.py", line 577, in run
    testMethod()
  File "/home/asamarchyan/projets/django_sfl/django/test/utils.py", line 377, in inner
    return func(*args, **kwargs)
  File "/home/asamarchyan/projets/django_sfl/tests/contenttypes_tests/test_checks.py", line 118, in test_generic_foreign_key_checks_are_performed
    checks.run_checks(app_configs=self.apps.get_app_configs())
  File "/home/asamarchyan/projets/django_sfl/django/core/checks/registry.py", line 78, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/home/asamarchyan/projets/django_sfl/django/contrib/auth/checks.py", line 103, in check_models_permissions
    Permission = apps.get_model('auth', 'Permission')
  File "/home/asamarchyan/projets/django_sfl/django/apps/registry.py", line 200, in get_model
    app_config = self.get_app_config(app_label)
  File "/home/asamarchyan/projets/django_sfl/django/apps/registry.py", line 156, in get_app_config
    raise LookupError(message)
LookupError: No installed app with label 'auth'.
```

File: tests/contenttypes_tests/test_checks.py (hunk 1 above)
Comment (desecho) [inline tests/contenttypes_tests/test_checks.py]: I removed it.

File: tests/contenttypes_tests/test_checks.py (hunk 2 above)
Comment (charettes) [inline tests/contenttypes_tests/test_checks.py]: I see, I suppose this check should be skipped when the `auth` app is not available. In the mean time feel free to ignore this comment.

**Pull request merged**